demos/local-only-todolist/README.md

```diff
@@ -0,0 +1,106 @@
+2. Enable user registration and sign in to the app.
+3. Notice how previously created data now seamlessly syncs with Supabase.
+
+For an overview of the mechanics behind this, see [the explanation further below](#how-this-works).
+
+# Run the app in local-only mode
+
+Ensure you have [melos](https://melos.invertase.dev/~melos-latest/getting-started) installed.
+
+1. `cd demos/local-only-todolist`
+2. `melos prepare`
+3. `flutter run`
+
+Create some data and notice how the data persists after a refresh or reopening the app.
+
+It is expected that users cannot register or sign in at this stage. This is because Supabase and PowerSync credentials have not been been configured yet -- there is a hardcoded check in `main.dart` for this. We'll get to this next.
+
+# Enable user registration and sign in
+
+Create Supabase and PowerSync projects, and add their credentials to `lib/app_config.dart` to enable user registration and sign in.
+
+## Set up your Supabase project
+
+Detailed instructions for integrating PowerSync with Supabase can be found in [the integration guide](https://docs.powersync.com/integration-guides/supabase). Below are the main steps required to get this demo running.
+
+Create a new Supabase project, and paste an run the contents of [database.sql](./database.sql) in the Supabase SQL editor.
+
+It does the following:
+
+1. Create `lists` and `todos` tables.
+2. Create a publication called `powersync` for `lists` and `todos`.
+3. Enable row level security, allowing users to only view and edit their own data.
+4. Create a trigger to populate some sample data when an user registers.
+
+## Configure your PowerSync Instance
+
+Create a new PowerSync instance, connecting to the database of the Supabase project. See instructions [here](https://docs.powersync.com/integration-guides/supabase-+-powersync#connect-powersync-to-your-supabase).
+
+Then deploy the following sync rules:
+
+```yaml
+bucket_definitions:
+  user_lists:
+    # Separate bucket per todo list
+    parameters: select id as list_id from lists where owner_id = request.user_id()
+    data:
+      - select * from lists where id = bucket.list_id
+      - select * from todos where list_id = bucket.list_id
+```
+
+## Configure the app
+
+Insert the credentials of your Supabase and PowerSync projects into `lib/app_config.dart`.
+
+## Sign in to the app
+
+Restart the app and sign up or sign in. Once successfully signed in, existing and new data should seamlessly sync with Supabase.
+
+
+# How this works
+
+This app uses [local-only](https://pub.dev/documentation/powersync/latest/powersync/Table/Table.localOnly.html) tables to persist data until the user has registered or signed in. Local-only tables do not log updates in an upload queue, avoiding any overhead or growth in database size.
+
+Once the user registers, the data is moved over to synced tables, at which point the data would be placed in the upload queue and will start syncing.
```

Comment:
Once the user registers, the data is moved over to synced tables. The move operation also places the data in the upload queue.